src/ctia/graphql/schemas.clj

```diff
@@ -42,137 +44,167 @@
+                      :args    (merge common/lucene-query-arguments
+                                      sighting/sighting-order-arg
+                                      p/connection-arguments)
+                      :resolve (res/search-entity-resolver :sighting)}
+   :target_record    {:type    TargetRecordType
+                      :args    search-by-id-args
+                      :resolve (res/entity-by-id-resolver :target-record)}
+   :target_records   {:type    TargetRecordConnectionType
+                      :args    (merge common/lucene-query-arguments
+                                      target-record/target-record-order-arg
+                                      p/connection-arguments)
+                      :resolve (res/search-entity-resolver :target-record)}
+   :tool             {:type    ToolType
+                      :args    search-by-id-args
+                      :resolve (res/entity-by-id-resolver :tool)}
+   :tools            {:type    ToolConnectionType
+                      :args    (merge common/lucene-query-arguments
+                                      tool/tool-order-arg
+                                      p/connection-arguments)
+                      :resolve (res/search-entity-resolver :tool)}
+   :vulnerability    {:type    VulnerabilityType
+                      :args    search-by-id-args
+                      :resolve (res/entity-by-id-resolver :vulnerability)}
+   :vulnerabilities  {:type    VulnerabilityConnectionType
+                      :args    (merge common/lucene-query-arguments
+                                      vulnerability/vulnerability-order-arg
+                                      p/connection-arguments)
+                      :resolve (res/search-entity-resolver :vulnerability)}
+   :weakness         {:type    WeaknessType
+                      :args    search-by-id-args
+                      :resolve (res/entity-by-id-resolver :weakness)}
+   :weaknesses       {:type    WeaknessConnectionType
+                      :args    (merge common/lucene-query-arguments
+                                      weakness/weakness-order-arg
+                                      p/connection-arguments)
+                      :resolve (res/search-entity-resolver :weakness)}})
+
+(defn- dash->underscore [k]
+  (-> k
+      name
+      (string/replace "-" "_")
+      keyword))
+
+(defn- disabled-entities
+  [{{:keys [enabled?]} :FeaturesService}]
+  (->> (entities/all-entities)
+       keys
+       (filter #(not (enabled? %)))))
+
+(defn- remove-disabled
+  "Removes GraphQl fields of keys of disabled entities."
+  [services graphql-fields]
+  (let [+plurals (reduce
+                  (fn [a n] (let [k      (dash->underscore n)
+                                  plural (-> k name (str "s") keyword)]
+                              (into a [k plural])))
+                  []
+                  (disabled-entities services))]
+    (apply dissoc graphql-fields +plurals)))
```

Comment:
considering my previous comment, I would write this function like this:
``` clojure
(->> (disabled-entities services)
     (mapcat (fn [{:keys [entity plural]}] [entity plural]))]
     (apply dissoc graphql-fields))
```

Reply:
A variant with `enbaled-entities` could be more elegant abd easier to reason about:
``` clojure
(->> (enabled-entities services)
     (mapcat (fn [{:keys [entity plural]}] [entity plural]))]
     (select-keys graphql-fields))
```